### ISSUE TITLE
htpasswd fails on state=absent on absent file

### ISSUE THREAD

**Issue by HanXHX** [ansible/ansible-modules-core#2619]
Hi,

Tested Ansible versions: 1.9.2 / 1.9.4
OS: Xubuntu

Test:

```
$ ansible -c local -i localhost, -m htpasswd -a 'name=foo password=bar state=absent path=test.htpasswd' localhost
localhost | FAILED >> {
    "failed": true,
    "msg": "test.htpasswd does not exists"
}
```

Expected:
- Status: OK
- Action: nothing to do

Thank-you!

Emilien

**Comment by robynbergeron** [comment]
[module: web_infrastructure/htpasswd.py]

**Comment by robynbergeron** [comment]
@HanXHX -- Thanks for submitting this bug report.

@ansible/core team -- FYI.

**Comment by abadger** [comment]
@HanXHX Thanks for reporting this.  Having something here is nice in that it catches typos but you're right, an error seems too strong.  In ansible-2.0 we have the ability to send warnings back from the remote machine to the controller.  I've just moved the error to a warning for htpassword in the devel and stable-2.0 branches.

Unfortunately, 1.9.x doesn't have the ability to display warnings from the modules so I can't do the same thing there.  So this will be fixed in ansible-2.0 when it comes out (Real Soon Now(TM) ) but behaviour won't be changed in 1.9.x.  

# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
